server/src/main/java/org/apache/druid/segment/indexing/DataSchema.java

```diff
@@ -41,13 +41,17 @@
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 
 /**
+ *
  */
 public class DataSchema
 {
   private static final Logger log = new Logger(DataSchema.class);
-
+  private static final Pattern INVALIDCHARS = Pattern.compile("(?s).*[^\\S ].*");
```

Comment:
Should we be checking for more invalid characters? Since the underlying issue was an unfriendly character causing issues with zookeeper paths, maybe we should forbid everything [that is not allowed in zookeeper paths](http://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html#ch_zkDataModel). I think these are the relevant ones:

>ZooKeeper has a hierarchal name space, much like a distributed file system. The only difference is that each node in the namespace can have data associated with it as well as children. It is like having a file system that allows a file to also be a directory. Paths to nodes are always expressed as canonical, absolute, slash-separated paths; there are no relative reference. Any unicode character can be used in a path subject to the following constraints:

> * The null character (\u0000) cannot be part of a path name. (This causes problems with the C binding.)
> * The following characters can't be used because they don't display well, or render in confusing ways: \u0001 - \u0019 and \u007F - \u009F.
> * The following characters are not allowed: \ud800 -uF8FFF, \uFFF0-uFFFF, \uXFFFE - \uXFFFF (where X is a digit 1 - E), \uF0000 - \uFFFFF.

This might